apps/passport-server/src/routing/server.ts

```diff
@@ -85,10 +85,28 @@ export async function startHttpServer(
           };
         }
 
+        // 86400 is the highest value supported by browsers for caching CORS
+        // responses
+        corsOptions.maxAge = 86400;
+
+        // Allows a later middleware to add additional caching headers
+        corsOptions.preflightContinue = true;
+
         callback(null, corsOptions);
       })
     );
 
+    // Add cache-control to preflight CORS responses in a separate middleware
+    // This is required for the browser to be able to cache these requests
+    app.use((req, res, next) => {
+      if (req.method === "OPTIONS") {
+        res.setHeader("Cache-Control", "public, max-age=86400");
+        res.end();
```

Comment:
slightly confused about the `res.end` here - from my understanding this middleware sits between the internet and the rest of the routes in this server - i presume there's some default handler for `OPTIONS` requests that the `res.end()` would prevent from being invoked, meaning that you're caching an empty response? Or am I missing something.

Reply:
The response is meant to be empty. The only thing a CORS response can really do is return `204` or some kind of error status. There is never a response body.

Reply:
Also, this middleware is only being called because `preflightContinue` was set to `true`. If it were left at the default `false` setting, the response would have ended with the previous middleware.